node-red-contrib-ms-language/node-luis.js

```diff
@@ -5,74 +5,72 @@ const extractEntities = (prediction) => {
+        .reduce((result, entity) => result.concat(...entity), []) // flat entity array
+        .forEach(entity => {
+            if (!entity) return;
+            if (entity.modelType === 'Composite Entity Extractor') {
+                compositeEntities.push({
+                    parentType: entity.type,
+                    value: entity.text,
+                    children: []
                 });
+                Object.values(prediction.entities[entity.type][0].$instance).forEach(child => {
+                    compositeEntities[compositeEntities.length - 1].children.push({
+                      type: child[0].type,
+                      value: child[0].text
+                    });
+                    entities.push({
+                      type: child[0].type,
+                      entity: child[0].text,
+                      startIndex: child[0].startIndex,
+                      endIndex: child[0].startIndex + child[0].length,
+                      score: entity.score
+                    })
+                  });
+            } else {
                 entities.push({
-                  type: child[0].type,
-                  entity: child[0].text,
-                  startIndex: child[0].startIndex,
-                  endIndex: child[0].startIndex + child[0].length,
-                  score: entity.score
-                })
-              });
-        } else {
-            entities.push({
-                type: entity.type,
-                entity: entity.text,
-                startIndex: entity.startIndex,
-                endIndex: entity.startIndex + entity.length, // add end index
-                resolution: {
-                  values: response.prediction.entities[entity.type.replace('builtin.', '')].reduce((result, e) => {
-                    if (Array.isArray(e)) {
-                      return result.concat(...e);
+                    type: entity.type,
+                    entity: entity.text,
+                    startIndex: entity.startIndex,
+                    endIndex: entity.startIndex + entity.length, // add end index
+                    resolution: {
+                      values: prediction.entities[entity.type.replace('builtin.', '')].reduce((result, e) => {
+                        if (Array.isArray(e)) {
+                          return result.concat(...e);
+                        }
+                        result.push(e);
+                        return result;
+                      }, [])
                     }
-                    result.push(e);
-                    return result;
-                  }, [])
-                }
-            });
-        }
-    });
+                });
+            }
+        });
+    }
 
     return {
         entities,
         compositeEntities
     };
 };
 
-const asyncGetPrediction = async function(node, text) {
-    let entityObject = {
-        entities: [],
-        compositeEntities: []
+const getPrediction = async function(node, text) {
+    const headers = {
```

Comment:
Here we should check if the node.subKey value is provided. In "endpoint" mode, this value is provided in the query parameters.
We should create an empty "headers" object and add the "Ocp-Apim-Subscription-Key" only when the value is defined.